refiner-lib/src/refiner_inner.rs

```diff
@@ -626,6 +663,17 @@ fn fill_tx(
         .s(U256::zero())
 }
 
+/// Try to extract an address from an execution outcome status.
+fn extract_address(status: &ExecutionStatusView) -> Result<Option<Address>, RefinerError> {
+    if let ExecutionStatusView::SuccessValue(value) = status {
+        base64::decode(value)
+            .map_err(RefinerError::SuccessValueBase64Args)
+            .map(|value| Address::try_from_slice(&value).ok())
```

Comment:
This implementation is correct for `InnerTransactionKind::DeployErc20` because it returns the address directly, but it is incorrect for the `Submit` and `Deploy` cases because they return the result wrapped in a `SubmitResult`.

Reply:
Will try to expand more here to check if I'm understanding correctly:

In case the 'InnerTransactionKind' is 'DeployErc20', then 'value' is a string from which the 'Vec<u8>' decoded has the bytes of an actual Address.

But in case the 'InnerTransactionKind' is 'Deploy' or 'SubmitResult', the 'value' is a string from which the 'Vec<u8>' decoded has the bytes of a Borsh-encoded 'SubmitResult' object. Giving this I will then need to Deserialize it to a 'SubmitResult' object and use the 'status: TransactionStatus' enum property to attempt to get the address from the 'Succeed(Vec<u8>)' enum value

Is this the correct interpretation?

Reply:
This conversation is not relevant now because of the use of the create_legacy_address function from engine.